content/en/docs/Tasks/creating-an-index.md

```diff
@@ -5,37 +5,100 @@ description: >
+OLM uses a plaintext [file-based catalog][file-based-catalog-spec] format (JSON or YAML) to store these records in an index, and `opm` has tooling
+that helps initialize an index, render new records into it, and then validate that the index is valid. Let's walk
+through a simple example.
 
->Note: The container image also contains information that represents the upgrade graphs between different operator versions, an operator's dependencies etc graphically. To learn more about the upgrade graph of an operator, checkout the [creating an upgrade graph doc][upgrade-graph-doc]  
+First, we need to initialize our index, so we'll make a directory for it, generate a Dockerfile that can build an index
+image, and then populate our index with our package definition.
 
-So, to make your operator available to OLM, you can generate an index image via opm with your bundle reference included:
 
+### Initialize the index
 ```sh
-$ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bundle:0.0.1 --tag quay.io/my-container-registry-namespace/my-index:1.0.0
-$ podman push quay.io/my-container-registry-namespace/my-index:1.0.0
+$ mkdir example-operator-index
+$ opm alpha generate dockerfile example-operator-index
+$ opm init example-operator \
+    --default-channel=preview \
+    --description=./README.md \
+    --icon=./example-operator.svg \
+    -output yaml > example-operator-index/index.yaml
 ```
 
-The resulting image is referred to as an `Index`. Now that image is available for clusters to use and reference with `CatalogSources` on their cluster.
+Let's validate our index to see if we're ready to ship!
+```sh
+$ opm validate example-operator-index
+FATA[0000] invalid index:
+└── invalid package "example-operator":
+    └── invalid channel "preview":
+        └── channel must contain at least one bundle
+```
+
+Alright, so it's not valid. It looks like we need to add a bundle, so let's do
+that next...
 
-`Index` images are additive, so you can add a new version of your operator bundle when you publish a new version of your operator:
+### Add a bundle to the index
 
-```bash
-$ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bundle:0.0.2 --from-index quay.io/my-container-registry-namespace/my-index:1.0.0 --tag quay.io/my-container-registry-namespace/my-index:1.0.1
+```sh
+$ opm render quay.io/example-inc/example-operator-bundle:v0.1.0 \
+    --output=yaml > example-operator-index/index.yaml
 ```
 
-### Other operations on an Index using `opm` 
+Let's validate again:
+```sh
+$ opm validate example-operator-index
+```
+
+Success! There were no errors and we got a `0` error code.
```

Comment:
super minor nit/can be ignored completely: add a `$ echo $?` to show there was a zero return code?